pkg/kubelet/kubelet.go

```diff
@@ -3669,3 +3750,60 @@ func (kl *Kubelet) shapingEnabled() bool {
 	}
 	return true
 }
+
+// killPodNow kills a pod.
+// The pod status is updated, and then it is killed with the specified grace period.
+// This function blocks until the pod is killed, an error is encountered, or a timeout is determined.
+// Arguments:
+// pod - the pod to kill
+// status - the desired status to associate with the pod (i.e. why its killed)
+// gracePeriodOverride - the grace period override to use instead of what is on the pod spec
+func (kl *Kubelet) killPodNow(pod *api.Pod, status api.PodStatus, gracePeriodOverride *int64) error {
```

Comment:
note to self: i had moved this into pod workers, and forgot to update when rebasing.  will fix up later.